immuni_exposure_ingestion/tasks/process_uploads.py

```diff
@@ -60,70 +66,154 @@ async def _process_uploads() -> None:
             )
-            last_index = 0
+            break
+        keys += extract_keys_with_risk_level_from_upload(upload)
+        processed_uploads.append(upload.id)
+
+    if (n_keys := len(keys)) > 0:
+        # Sort the keys. This randomizes their order (since they are random strings) so that
+        # keys of the same device are no more likely to end up consecutively.
+        keys = sorted(keys, key=lambda x: x.key_data)
+
+        index = last_index + 1
+
+        batch_file = BatchFile(
+            index=index,
+            keys=keys,
+            period_start=period_start,
+            period_end=period_end,
+            sub_batch_index=1,
+            sub_batch_count=1,
+        )
+        batch_file.client_content = batch_to_sdk_zip_file(batch_file)
+        batch_file.save()
+        _LOGGER.info("Created new italian batch.", extra=dict(index=index, n_keys=n_keys))
+        BATCH_FILES_CREATED.inc()
+        KEYS_PROCESSED.inc(len(keys))
 
-        period_start = last_period
-        period_end = now
+    Upload.set_published(processed_uploads)
+    _LOGGER.info(
+        "Flagged italian uploads as published.",
+        extra=dict(n_processed_uploads=len(processed_uploads)),
+    )
+    UPLOADS_ENQUEUED.set(Upload.to_process().count())
 
-        _LOGGER.info(
-            "Starting to process uploads.",
-            extra=dict(period_start=period_start, period_end=period_end),
-        )
 
-        uploads = Upload.to_process()
-
-        _LOGGER.info("Uploads have been fetched.", extra=dict(n_uploads=uploads.count()))
-
-        processed_uploads: List[ObjectId] = []
-        keys: List[TemporaryExposureKey] = []
-        for upload in uploads:
-            if (reached := len(keys) + len(upload.keys)) > config.MAX_KEYS_PER_BATCH:
-                _LOGGER.warning(
-                    "Early stop: reached maximum number of keys per batch.",
-                    extra=dict(
-                        pre_reached=len(keys), reached=reached, max=config.MAX_KEYS_PER_BATCH
-                    ),
-                )
-                break
-            keys += extract_keys_with_risk_level_from_upload(upload)
-            processed_uploads.append(upload.id)
-
-        if (n_keys := len(keys)) > 0:
-            # Sort the keys. This randomizes their order (since they are random strings) so that
-            # keys of the same device are no more likely to end up consecutively.
-            keys = sorted(keys, key=lambda x: x.key_data)
-
-            index = last_index + 1
-
-            batch_file = BatchFile(
-                index=index,
-                keys=keys,
-                period_start=period_start,
-                period_end=period_end,
-                sub_batch_index=1,
-                sub_batch_count=1,
+def _batch_eu():
+    """
+    Get the unprocessed uploads from the upload_eu collection
+    having also the country parameter set to "IT",
```

Comment:
Let's try to go new line close to, but before, 100 chars. This keeps the documentation consistent with the code. 🙏

Reply:
Also, I'd improve the docstring with a concrete example, like the one you gave me over the call. 🙏